### ISSUE TITLE
Dématérialisation des autorisations d’urbanisme

### ISSUE THREAD

**Issue by SDIS91** [SDIS62/prevarisc#695]
Une évolution règlementaire éminemment cruciale a fait convoquer les éditeurs de logiciels de gestion Prévention par la DGSCGC pour visiblement l’intégrer à leurs solutions. Est-ce prévu dans PREVARISC ? Sera-t-elle intégrée par l’un ou l’autre des SDIS utilisateurs ?

Voir
[https://www.cohesion-territoires.gouv.fr/dematerialisation-des-autorisations-durbanisme](https://www.cohesion-territoires.gouv.fr/dematerialisation-des-autorisations-durbanisme)

Merci

**Comment by kdubuc** [comment]
Oui, nous travaillons déjà dessus, et nous communiquerons très prochainement sur le développement de la passerelle de connexion entre Plat'Au et Prevarisc ! L'objectif étant d'avoir une passerelle fonctionnelle d'ici janvier 2021.

**Comment by sdis12** [comment]
Super nouvelle si les développements se poursuivent. 
Le code source de prévarisc n'a pas été mis à jours depuis plusieurs années et certains bug d'utilisations persistentent (je pense coté carto notamment et certainement d'autres). Est il prévu par le SDIS62 de mettre à disposition les différentes évolutions/correction de bug ?

Merci pour votre retour.
le SDIS12.

**Comment by kdubuc** [comment]
Effectivement, le dépôt Prevarisc est en sommeil depuis quelques années. Il est difficile de libérer des moyens pour maintenir et entretenir le code source public. Nous pourrions aussi mieux faire au niveau de la gestion de la communauté utilisateur, c'est un constat que je partage malgré le fait que ce soit assez compliqué à gérer depuis un SDIS d'avoir ce rôle de référent national.

Néanmoins, nous travaillons activement depuis quelques mois avec certains interlocuteurs afin de fluidifier et simplifier les deux points évoqués ci dessus. J'espère pouvoir vous annoncer cela en début d'année 2021.

**Comment by sdis12** [comment]
Je comprend effectivement la problématique. En espérant que ça va s'améliorer. En tout cas c'est encourageant de savoir que ce sujet n'est pas laissé à l'abandon !
Merci pour le retour

**Comment by SDIS91** [comment]
Hello

Des nouvelles de la passerelle de connexion entre Plat'Au et Prevarisc?

Merci

**Comment by kdubuc** [comment]
La passerelle passe les tests du programme Plat'AU la semaine prochaine. Si tout se passe bien, mise en ligne vendredi ! ;)

**Comment by SDIS91** [comment]
Merci beaucoup

**Comment by kdubuc** [comment]
La première version de la passerelle est disponible ici : https://github.com/SDIS62/prevarisc-passerelle-platau 🎆 

Un patch pour Prevarisc sera déployé dans la journée pour rendre effective la communication entre Prevarisc et la passerelle.

**Comment by kdubuc** [comment]
Voila le commit / patch que vous devez appliquer avant d'utiliser la passerelle : https://github.com/SDIS62/prevarisc/commit/b9626ba37b591d251233d071490fcff0f44420ff

Je reste dispo ici (ou sur le dépôt de la passerelle directement) pour vous expliquer comment fonctionne l'ensemble, n'hésitez pas !

**Comment by SDIS33** [comment]
> Voila le commit / patch que vous devez appliquer avant d'utiliser la passerelle : [b9626ba](https://github.com/SDIS62/prevarisc/commit/b9626ba37b591d251233d071490fcff0f44420ff)
> 
> Je reste dispo ici (ou sur le dépôt de la passerelle directement) pour vous expliquer comment fonctionne l'ensemble, n'hésitez pas !

Merci pour ce travail et le partage.

Nous sommes en train d'intégrer la passerelle Plat'AU au SDIS33, il nous manque toutefois quelques informations pour la rendre fonctionnelle, pouvez-vous nous indiquer :
- si le code organisation à saisir lors de l'inscription à PISTE est bien 984b19, ou si chaque établissement en a un ? Dans ce cas, comment obtenir ce code organisation ?
- comment obtenir l'identifiant Acteur Plat'AU ?

En vous remerciant par avance.

**Comment by kdubuc** [comment]
> si le code organisation à saisir lors de l'inscription à PISTE est bien 984b19, ou si chaque établissement en a un ? Dans ce cas, comment obtenir ce code organisation ?

Ce code organisation correspond à l'environnement MTES. Seulement, cet environnement correspond à la production, les tests se déroulants dans un autre environnement. Une présentation des environnements est disponible sur le support SPS Plat'AU.

> comment obtenir l'identifiant Acteur Plat'AU ?

De la même façon, en passant par le support SPS Plat'AU, vous pouvez y poster un ticket de demande d'ajout d'acteurs.

**Comment by SDIS33** [comment]
Bonsoir,

Après enfin avoir réussi à créer un acteur pour pouvoir utiliser la passerelle, nous souhaitons tester son utilisation.

En voulant installer la passerelle sur un serveur Prevarisc de recette, je me suis aperçu que celle-ci n'était pas compatible avec la version de PHP installée sur celui-ci (5.6).
Par ailleurs, je n'ai pas pu être présent à la visio organisée en début de semaine sur le sujet, mais on m'a remonté l'information qu'il serait préférable d'installer la passerelle sur un autre serveur que Prevarisc.

Pouvez-vous m'en expliquer la raison ?
Pouvez-vous également m'indiquer les prérequis techniques à l'installation de celle-ci (version de PHP conseillée, etc...) ?

Merci d'avance et bon week-end.

**Comment by dtenet** [comment]
Prevarisc étant une application dont le développement date de 2014, il se base sur un socle PHP 5. En 2021, quand la passerelle Plat'AU a été développé, elle est basée sur un socle beaucoup plus récent, un PHP 7.4 (pour diverses raisons techniques qui seraient trop longue à expliquer ici).

Vous avez donc deux possibilités:
1/ faire cohabiter deux versions de php différentes sous le même serveur et installer les deux applications sur le même serveur.
2/ faire deux serveurs séparés avec chacun sa version de PHP, un en 5.X l'autre en 7.4 et une application par serveur.

Pour pouvoir installer une version de PHP dans une certaine version, il y a des pré-requis (dépendance avec d'autres paquetages annexes) et plus vous cherchez à installer une vieille version de PHP sur un serveur Linux récent et plus c'est compliqué. L'inverse (installer une version PHP sur un linux ancien) est également assez ardu. L'idée de séparer les deux applications sur deux serveurs est finalement peut être la plus simple à maintenir suivant votre niveau d'expertise.

**Comment by SDIS33** [comment]
> Prevarisc étant une application dont le développement date de 2014, il se base sur un socle PHP 5. En 2021, quand la passerelle Plat'AU a été développé, elle est basée sur un socle beaucoup plus récent, un PHP 7.4 (pour diverses raisons techniques qui seraient trop longue à expliquer ici).
> 
> Vous avez donc deux possibilités: 1/ faire cohabiter deux versions de php différentes sous le même serveur et installer les deux applications sur le même serveur. 2/ faire deux serveurs séparés avec chacun sa version de PHP, un en 5.X l'autre en 7.4 et une application par serveur.
> 
> Pour pouvoir installer une version de PHP dans une certaine version, il y a des pré-requis (dépendance avec d'autres paquetages annexes) et plus vous cherchez à installer une vieille version de PHP sur un serveur Linux récent et plus c'est compliqué. L'inverse (installer une version PHP sur un linux ancien) est également assez ardu. L'idée de séparer les deux applications sur deux serveurs est finalement peut être la plus simple à maintenir suivant votre niveau d'expertise.

OK, merci pour l'explication.

Je suis également preneur de tout autre conseil d'ordre technique (placer la passerelle sur le LAN ou en DMZ ? ...), mais aussi d'informations plus précises quant au fonctionnement des 3 cas d'utilisation de la passerelle, notamment les règles de gestion qui concernent les intéractions entre PREVARISC et PLAT'AU (ATOS avait indiqué qu'il partagerait une doc fonctionnelle sur le sujet).

**Comment by valentinrouxelrennes** [comment]
Bonjour, la documentation a été déposée sur le lien suivant :

https://github.com/atos-df-rennes/prevarisc-passerelle-platau/blob/768d1be0d563c8214d6ff9cc75aaa2075514329e/Manuel%20Utilisateur%20Plat'AU.pdf

**Comment by dtenet** [comment]
Pour le placement de la passerelle Plat'AU au niveau du réseau, il est compliqué de faire une réponse type. Il faut prendre en compte les éléments de votre infra. Il faut savoir que Plat'AU doit discuter avec la base de données de Prevarisc ET être capable de joindre une adresse sur internet pour dialoguer avec Plat'AU. Sur le papier, le placement du serveur passerelle prevarisc en DMZ est donc pertinent mais le serveur passerelle prevarisc initie toujours la connexion et ne doit pas forcément recevoir des connexions entrantes sans qu'il ne l'est initié au départ. Il peut donc rester au même niveau que votre serveur Prevarisc. A adapter selon vos politiques de sécurité internes.

**Comment by SDIS33** [comment]
Super, merci pour les infos !

**Comment by dellac** [comment]
La demat sur la rampe de lancement à SDIS 33 !! Merci CFX pour l'atterrissage des premiers dossiers. 
N'hésitez pas à échanger vos expériences connexion Prevarisc Plat'Au , elle seront certainement très variées et instructives.
Je suis intéressé par vos avis sur le nom des pièces jointes reçues. Exploitables ou à renommer ?
Meilleurs voeux à toutes et tous

**Comment by dellac** [comment]
@kdubuc  
Demande de confirmation
L'avis sur étude est il bien rendu lorsque les 2 conditions suivantes sont remplies simultanément ? :
- dossier complet ou incomplet,
- avis commission favorable ou défavorable.

Si oui, y a t'il la possibilité actuellement qu'une pièce jointe, le rapport d'étude, soit envoyée automatiquement au rendu de l'avis ?

**Comment by SDIS33** [comment]
Bonjour,

Comme indiqué par mon collègue, nous avons pu récupérer les premiers dossiers provenant de Plat'AU grâce à votre passerelle et vos indications, merci encore.

Toutefois, nous rencontrons encore un problème en ce qui concerne la récupération des pièces jointes.
Pouvez-vous nous confirmer que ce mécanisme est bien inclus dans la fonctionnalité d'import de la passerelle ?

En regardant un peu le code de la passerelle, j'ai bien identifié la fonction creerPieceJointe dans le fichier Service/Prevarisc.php, mais j'ai l'impression qu'elle n'est jamais appelée...

D'autre part, je me pose une question quant au fonctionnement de l'écriture sur le serveur des pièces jointes récupérées : 
- on indique bien le chemin d'écriture de ces PJ par le biais de la propriété "PREVARISC_PIECES_JOINTES_PATH" du fichier JSON de configuration de la passerelle ;
- j'imagine que ce chemin correspond au chemin utilisé par PREVARISC pour gérer ses pièces jointes sur son serveur (chez nous "/var/www/prevarisc/public/data/uploads/pieces-jointes"), mais cela ne pose-t-il pas un problème si on a installé la passerelle sur un autre serveur ? Faut-il configurer une connexion ssh par clé publique ou autre ?

Merci d'avance et bon week-end.

**Comment by sdis12** [comment]
Bonjour,

Je pense qu'il vous faut poser plus les question sur la composant platau prevarisc de github  (https://github.com/SDIS62/prevarisc-passerelle-platau/issues) et non ici.
Ceci dit quelques éléments mais n'ayant pas de dossier de notre cote pour tester ce ne sont que des suppositions. Effectivement le variable  PREVARISC_PIECES_JOINTES_PATH du fichier Json doit correspondre au dossier où sont stockés les pièces jointes de prévarisc.
Si vous êtes dans une configuration en serveur séparé il faut donc que votre serveur platau puisse accéder à ce dossier. Pour cela tout depend ou sont stocké vos piece jointes actuellement. Si c'est en local sur votre serveur prevarisc il faudra effectivement que le serveur Platau puisse monter le dossier des pièces jointes de prevarisc. Pour un montage entre 2 serveur linux, NFS semble etre une bonne solution native mais il existe plein de façon de le faire. 

Cordialement,
